url.c

```diff
@@ -96,20 +96,72 @@ enum UrlScheme url_check_scheme(const char *s)
     return (enum UrlScheme) i;
 }
 
+
+int _parse_query_string(struct Url *u, char *src)
+{
+  struct UrlQueryString *qs = NULL;
+  char *k = NULL, *v = NULL;
+
+  while (src && *src)
+  {
+    qs = safe_calloc(1, sizeof(struct UrlQueryString));
+    if ((k = strrchr(src, '&')))
+      *k = '\0';
+
+    if ((v = strchr(src, '=')))
+    {
+      *v = '\0';
+      qs->value = v + 1;
+      if (url_pct_decode(qs->value) < 0)
+        return -1;
+    }
+    qs->name = src;
+    if (url_pct_decode(qs->name) < 0)
+      return -1;
```

Comment:
... and here.

Reply:
Done, and I also freed the query_strings when in url_parse() return -1.